spark-mongodb/src/main/scala/com/stratio/datasource/mongodb/MongodbRelation.scala

```diff
@@ -55,7 +55,7 @@ with PrunedFilteredScan with InsertableRelation {
   @transient private lazy val lazySchema =
     MongodbSchema(
       new MongodbRDD(sqlContext, config, rddPartitioner),
-      config.getOrElse[Double](MongodbConfig.SamplingRatio, MongodbConfig.DefaultSamplingRatio)).schema()
+      config.get[String](MongodbConfig.SamplingRatio).map(_.toDouble).getOrElse(MongodbConfig.DefaultSamplingRatio)).schema()
```

Comment:
Instead of map followed by getOrElse, you can use the fold function.

``` config.get[String](MongodbConfig.SamplingRatio).fold(MongodbConfig.DefaultSamplingRatio)(_.toDouble).schema() ```